### ISSUE TITLE
You might be interested in this! 🙂 

### ISSUE THREAD

**Issue by mithi** [r002/captains-log#50]
Hi @r002 !

I don't know if you remember me, but you submitted a pull request in my repo [mithi/epic-react-exercises](http://github.com/mithi/epic-react-exercises). We had chats about different coding philosophies and ideas about separation of concerns and typescript..
I didn't know how to contact you, so I'm just submitting an issue here. 😅 
I'm writing because I wrote something that you might be interested in, it's a short document where I gather my thoughts about writing good React code. I hope you like it, and if you have any comments, suggestions or violent reactions, don't hesitate to tell me. Thanks! 

https://github.com/mithi/react-philosophies

**Comment by r002** [comment]
Hi @mithi!  Of course I remember you!  Your repo was the first public PR I **_ever_** committed to so of course you'll always have a special place in my heart!  🙂  Since then, I went on an open source streak committing to other projects from all over but with `Mithi's Epic React Exercises` was how it all began!  🥳

This `react-philosophies` document looks amazing though!  I read it quickly just now and am a big fan of everything I saw so far.  Especially the part about "the best code is no code!" 😄  This weekend I'll look it over more thoroughly and let you know more detailed thoughts though!  Thanks for putting this together-- you make the world and web a better place!!

Hope you're having a most wonderful day!  🎉🚀

**Comment by mithi** [comment]
Thanks @r002  🙂

**Comment by r002** [comment]
Hi @mithi!  Sorry for the delay.  But I finally just got time to properly review your `react-philosophies` document today.  It's a great writeup and I really think you did a terrific job!  👍  In all honesty, I'm more of a hobbyist coder these days and enjoy just perusing the tech landscape just to (well, try at least) "stay in the know".  Honestly, the JS space has exploded so ridiculously and I have a ton of trouble following it all.

Reading your doc really got me to thinking about a larger question though and I'm curious about your thoughts:  Do you think `react` is passé nowadays?  I know it's still super-popular but from my experience using it (again, not professionally or thoroughly, to be completely fair), but I really disliked it once I started using it on larger projects.  For example, I agree with you idea of `co-locating state` and keeping state as close to the component actually using it as possible.  But in my experience, whenever I added features to my website, I often found myself having to move state around which was just a huge pain.  Maybe I'm going it wrong but have you seen [the Richard Harris talk on Svelte](https://www.youtube.com/watch?v=AdNJ3fydeao)?  I loved his points and `Svelte` is the framework I'm looking at diving into next.  Any thoughts?

**Comment by mithi** [comment]
yes I love svelte! however svelte is not yet that mature.. the community is
very small compared to React and not a lot of companies are using svelte in
production for complex applications yet! i havent use svelte outside of
hobby toy projects to i dont know yet if it’s hard to create complex
interactive applications with it. I’m not sure if there are some features
that you can implement in React that is not so easy to implement in Svelte..

Im using react because professionally because that’s what the market
demands (there are many jobs ) and the react ecosystem is also great (there
are many well maintained libraries and tools with great developer
experience and API which you can confidently use for specific needs, no
need to implement components  from scratch!)

I believe React isnt going away any time soon because there are a lot of
very large complex applications that are built in React and I dont think
it’s just that easy to rewrite them is svete  😉

I’m not particularly attached to any framework or technologies when it
comes to the front end. I’m just interested in building  maintainable user
interfaces with amazing user experiences.

So bottomline, i’ll use whatever gets the job done. :) I’m willing to jump
ship anytime 😉
